src/main/groovy/com/twcable/gradle/cqpackage/CqPackageUtils.groovy

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright 2015 Time Warner Cable, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twcable.gradle.cqpackage
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.DependencySet
+import org.gradle.api.artifacts.ProjectDependency
+import org.gradle.api.internal.artifacts.CachingDependencyResolveContext
+import org.gradle.api.internal.artifacts.ResolvableDependency
+
+import javax.annotation.Nonnull
+import java.util.regex.Pattern
+
+import static java.util.Collections.EMPTY_LIST
+import static java.util.Collections.EMPTY_SET
+
+@Slf4j
+@CompileStatic
+class CqPackageUtils {
+    private static final Pattern JAR_NAME_PATTERN = Pattern.compile(/(?i).*\.jar$/)
+
+    /**
+     * Returns all JAR files generated by the given project and its project dependencies in
+     * the configuration that use the "osgi" plugin.
+     */
+    @Nonnull
+    static Collection<File> projectBundleJarFile(@Nonnull Project project, @Nonnull Configuration configuration) {
+        if (configuration == null) throw new IllegalArgumentException("configuration == null")
+
+        Collection<ProjectDependency> allOsgiProjects = projectDependenciesForConfiguration(configuration).
+            findAll { isOsgiProject(it.dependencyProject) }
+
+        log.debug "osgi projects: ${allOsgiProjects}"
```

Comment:
Wondering about two different strategies to log the message `project.logger.debug`.
https://github.com/TWCable/cq-gradle-plugin/pull/21/files#diff-3562802a1605cf74e30e672f3dcebdd2R94